### PR TITLE
Updates to metric compare table (extended names, population)

### PR DESCRIPTION
**Pull request by mikelehen**
* Add metric extended names to metric columns.
* Add population to location column.

To be honest, I am not sure it looks great (especially with the long metric names we have in storybook), but I think this roughly matches the mocks:

image

### CODE REVIEW

File: packages/ui-components/src/common/utils/index.ts [hunk 1]
```diff
@@ -1 +1,18 @@
...
+ * ```ts
+ * formatPopulation(107766) // 108,000
+ * ```
+ */
+export function formatPopulation(population: number) {
+  return formatDecimal(population, {
+    maximumSignificantDigits: 3,
+    maximumFractionDigits: 0,
+  });
+}
```
Comment (mikelehen) [inline packages/ui-components/src/common/utils/index.ts:17]: I moved `formatPopulation()` from `RegionSearch` to here.  I'm not sure I'm wild about putting it in the `index.ts` file though.  Maybe I should create a `misc.ts` file or something?

I could also move this into the number-format package, but I assume that's for general-purpose number-formatting, not specific kinds of numbers (like populations).

Comment (pnavarrc) [reply packages/ui-components/src/common/utils/index.ts:17]: Maybe we could have a second `formatInteger` function in `number-format` that takes options, so we can use currency symbols, thousands separator, etc

**Review by pnavarrc — APPROVED** [review]
Looks good!

File: packages/ui-components/src/common/utils/index.ts (hunk 1 above)
Comment (mikelehen) [reply packages/ui-components/src/common/utils/index.ts:17]: I went ahead and did this, but I also kept `formatPopulation()` to ensure that RegionSearch and CompareTable use the same logic. 🤷

**Pull request merged**